### DISCUSSION TITLE
Manual test start/stop

### DISCUSSION THREAD

**Discussion by pkrukp** [OpenCover/opencover#1026] [Ideas]
Hello,

How hard would it be to add possibility to manually start/stop tests?

1. Consider following use case: manual tests on a running application/web site. User have scenarios to execute (let's say QA tests) and wants to collect coverage for each scenario. So he'd like to start application, "start test", execute manual test (interact with app), "stop test" with some outcomes (failed/passed/etc).

2. Another use case: setting test name before starting application. For example you have a set of scripts/tests (integration/black box testing) that execute application. User would like to collect coverage for each test separately, associate the coverage with a test, and merge it together. On final merged report he would like to have information about each test and it's coverage.

---

I guess opencover would have to expose some api to enable it? Then when user starts a test it'd need to clear/reset coverage, and when user stop tests it'd have to collect coverage...

This is already kind of implemented - opencover can collect coverage per test ("-coverbytest"), but right now it somehow automatically detects the test executed, so instead of auto-detection it would have to let user define the test (and do it on "live" application too).

**Comment by sawilde** [comment]
I suppose it could be done but it may be simpler to stop and start opencover with each test run and capture data that way in different files.
